### PR TITLE
Documentation of Gtid/GtidSet

### PR DESCRIPTION
**Pull request by darnuria**
Add some documentation to Gtid/Gtidset. Most notable contribution documentation in detail of the binary format.

### CODE REVIEW

File: pymysqlreplication/gtid.py [hunk 1]
```diff
@@ -204,6 +210,14 @@ def encoded_length(self):
...
 
     def encode(self):
+        """Encode a Gtid in binary
+
+        Format:
+
+        - sid will be uncoded as hex-binary without the dashes,
+        - size of the interval list as a u64
+        - all the interval as a pair of u64
+        """
```
[the remark this answers is not in the document]
Comment (julien-duponchelle) [reply pymysqlreplication/gtid.py:220]: Oh that will be beautiful

File: pymysqlreplication/gtid.py [hunk 2]
```diff
@@ -204,6 +210,14 @@ def encoded_length(self):
                 len(self.intervals))
```
Comment (darnuria) [inline pymysqlreplication/gtid.py:210]: I am wondering what is this `2 *  # stop/start` in binary encoding and why only `8` for stop/start.

- [x] SID ok
- [x] interval len
- [ ]  2 *  # stop/start ??
- [ ] stop / start as 2 u64 OK


later we do for `start/stop`  reading: `start, end = struct.unpack('<QQ', payload.read(16))`

and writting is

```
        buffer += struct.pack('<Q', len(self.intervals))
        for interval in self.intervals:
            # Start position
            buffer += struct.pack('<Q', interval[0])
            # Stop position
            buffer += struct.pack('<Q', interval[1])
```

I am struggling with this 2 and 8. Any clue? If it's a mistery I can write a test XD

I have not seen explicit testing of `Gtid.encoded_length`.

File: pymysqlreplication/gtid.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (darnuria) [reply pymysqlreplication/gtid.py:220]: something like:

```
    /// Warning: Subject to change
    ///
    /// - `sid`: `16 * u8` as a `[u8;16]`
    /// - `intervals_len`: `8 * u8` as a u64
    ///
    /// `interval_len` times:
    ///     - `start`: `8 * u8` as `u64` start of interval
    ///     - `end`: `8 * u8` as u64 end of interval
    /// ```txt
    /// Alligned on u64 bit.
    /// +-+-+-+-+-+-+-+-+-+-+
    /// | sid [16;u8]       |
    /// |                   |
    /// +-+-+-+-+-+-+-+-+-+-+
    /// | intervals_len u64 |
    /// +-+-+-+-+-+-+-+-+-+-+
    /// |start u64              \
    /// - - - - - - - - - - -    + Repeated
    /// |stop u64               /  interval_len
    /// - - - - - - - - - - -      times
    ```

And 
```
    ///
    /// - `n_sid`: u64 is the number of Gtid to read
    /// - `Gtid`: `n_sid` * `Gtid_encoded_size` times See [Gtid] documentation for details.
    /// ```txt
    /// Alligned on u64 bit.
    /// +-+-+-+-+-+-+-+-+-+-+
    /// | n_gtid u64        |
    /// |                   |
    /// +-+-+-+-+-+-+-+-+-+-+
    /// | Gtid              | - Repeated n_gtid
    /// - - - - - - - - - - -   times
    /// ```
```
Would do the trick? it's what I intend to do in my rust crate only decoding gtid (licence are compatibles) if it sound Ok I can contribute it here!

**Comment by darnuria** [comment]
@julien-duponchelle let me know if the diagrams are not like you expect them, I am not a "PRO" with binary datagram. ;p

**Comment by julien-duponchelle** [comment]
That's great !

**Pull request merged**